### PR TITLE
OCM-2864 | feat: Added default push request template

### PR DESCRIPTION
**Pull request by den-rgb**
Added a new template for merge request's on the ROSA repo.

[OCM-2864](https://issues.redhat.com/browse/OCM-2864)

example of what it will look like :

### JIRA link

Please include a link to the JIRA

### Design document

Please include a link to the design document if there is one 
associated with these changes

### What's new?

Please describe the changes in this PR

### How to test?

Please describe how to test these changes

## PR Author Check List

* [ ]  All commit messages adhere to the project [standard](https://github.com/openshift/rosa/blob/master/CONTRIBUTING.md) 

* [ ]  All commits are squashed into a single commit `git rebase -i HEAD~<number of commits>`

* [ ]  All code changes have unit test coverage

* [ ]  All changes have been tested locally and the steps documented in the `How to Test?` section

* [ ]  All required API changes are deployed to production

### CODE REVIEW

**Comment by den-rgb** [comment]
> LGTM but could I suggest something like:
> 
>     * [ ]  has X tests
> 
> 
> i.e. unit tests
> 
> I think it would be useful but maybe this is what you were envisioning with the `How to test?` section 🙂

I think a step by step list off bullet points pointing out how to achieve the result of the changes would be helpful especially for new team members, this of course wouldn't be applicable to every change but for major changes that have prerequisites it would be nice, in regards to referencing which tests were ran , that's something i can add

**Comment by hunterkepley** [comment]
> I think a step by step list off bullet points pointing out how to achieve the result of the changes would be helpful 

+1

**Comment by robpblake** [comment]
@den-rgb Thanks for adding this 👍 

I personally find the self-review checklist item unnecessary and would trust that people are opening PRs when they think they're good for review. 

I'm also wondering if we can make the checklist a bit easier to action/process and help to enforce standards, something like:

## PR Author Check List

- [ ] All commit messages adhere to the project standard (link to CONTRIBUTING.md to show our standards)
- [ ] All commits are squashed into a single commit (link to CONTRIBUTING.md to show how to do this)
- [ ] All code changes have unit test coverage (missing coverage is explained)
- [ ] All changes have been tested locally and the steps documented in the `How to Test?` section
- [ ] All required API changes are deployed 

On the point of deployment, is the requirement that API changes are deployed to `production` before we can merge changes into this repo?

cheers,

Rob

**Comment by den-rgb** [comment]
> @den-rgb Thanks for adding this 👍
> 
> I personally find the self-review checklist item unnecessary and would trust that people are opening PRs when they think they're good for review.
> 
> I'm also wondering if we can make the checklist a bit easier to action/process and help to enforce standards, something like:
> ## PR Author Check List
> 
>     * [ ]  All commit messages adhere to the project standard (link to CONTRIBUTING.md to show our standards)
> 
>     * [ ]  All commits are squashed into a single commit (link to CONTRIBUTING.md to show how to do this)
> 
>     * [ ]  All code changes have unit test coverage (missing coverage is explained)
> 
>     * [ ]  All changes have been tested locally and the steps documented in the `How to Test?` section
> 
>     * [ ]  All required API changes are deployed
> 
> 
> On the point of deployment, is the requirement that API changes are deployed to `production` before we can merge changes into this repo?
> 
> cheers,
> 
> Rob

I'll implement that suggestion it seems much better then the above, and yes

File: .github/merge_request_template/Default.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+## Describe your changes
+
+### JIRA link
+<!--- Please include a link to the JIRA ticket if there is one associated with -->
```
Comment (ciaranRoche) [inline .github/merge_request_template/Default.md:4]: NP : every PR requires a JIRA ticket, I would expand this to point a user to the contributing guide if they do not have a JIRA ticket for their change

File: .github/merge_request_template/Default.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+### Design document
```
Comment (ciaranRoche) [inline .github/merge_request_template/Default.md:6]: I think we can leave this out, we have it in the API for context of folks outside of the squad, but I feel everyone in the squad should be familiar with the features and their design

File: .github/merge_request_template/Default.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+* [ ]  All commit messages adhere to the project [standard](https://github.com/openshift/rosa/blob/master/CONTRIBUTING.md) 
+
+* [ ]  All commits are squashed into a single commit `git rebase -i HEAD~<number of commits>`
+
+* [ ]  All code changes have unit test coverage
+
+* [ ]  All changes have been tested locally and the steps documented in the `How to Test?` section
+
+* [ ]  All required API changes are deployed to production
```
Comment (ciaranRoche) [inline .github/merge_request_template/Default.md:26]: We do not require this, this is a requirement of the Release, not merging to master

**Comment by robpblake** [comment]
@den-rgb What do you think about making the section on the commit message part of the `CONTRIBUTING.md` a header so that it can be linked to and is easier to find when reading the Contributing guide?

**Comment by den-rgb** [comment]
> @den-rgb What do you think about making the section on the commit message part of the `CONTRIBUTING.md` a header so that it can be linked to and is easier to find when reading the Contributing guide?

The commit message part is the first thing you see in the contribution guide anyway unless I misunderstand what you're referring to?

**Review by ciaranRoche — APPROVED** [review]
/lgtm

**Pull request merged**